src/ripple/app/misc/Validations.cpp

```diff
@@ -458,20 +467,25 @@ class ValidationsImp : public Validations
 
         mWriting = true;
         app_.getJobQueue ().addJob (
-            jtWRITE, "Validations::doWrite",
-            [this] (Job&) { doWrite(); });
+            jtWRITE, "Validations::queuedWrite",
+            [this] (Job&) { queuedWrite(); });
     }
 
-    void doWrite ()
+    void queuedWrite ()
     {
         auto event = app_.getJobQueue ().getLoadEventAP (jtDISK, "ValidationWrite");
 
-        std::string insVal ("INSERT INTO Validations "
+        ScopedLockType sl (mLock);
+        doWrite (sl);
+    }
+
+    void doWrite (ScopedLockType& sl)
```

Comment:
Can you explain the purpose of passing in the scoped lock to `doWrite`?  I'm not able to determine why its needed.